resources/assets/js/store/Application/applicationSelector.ts

```diff
@@ -39,6 +39,23 @@ const constructNonNormalizedApplication = (
   };
 };
 
+export const getApplicationIsUpdating = (
+  state: RootState,
+  props: { applicationId: number },
+): boolean =>
+  hasKey(ui(state).applicationIsUpdating, props.applicationId) &&
+  ui(state).applicationIsUpdating[props.applicationId];
+
+export const getApplication = createCachedSelector(
+  getApplicationState,
+  (state: RootState, ownProps: { applicationId: number }): number =>
+    ownProps.applicationId,
+  (applicationState, applicationId): ApplicationNormalized | null =>
+    hasKey(applicationState, applicationId)
+      ? applicationState[applicationId]
+      : null,
+)((state, ownProps): number => ownProps.applicationId);
```

Comment:
I'm not sure this selector gains anything from using (re-)reselect. As far as I know, the main reason to cache selectors (besides when you have a computationally expensive selector) is to avoid returning something like a new array object (which will cause re-renders) when the contents are the same.